examples/bootloader/guiFunctions.py

```diff
@@ -0,0 +1,228 @@
+        return dai.UsbSpeed.LOW
+    elif s == "FULL":
+        return dai.UsbSpeed.FULL
+    elif s == "HIGH":
+        return dai.UsbSpeed.HIGH
+    elif s == "SUPER":
+        return dai.UsbSpeed.SUPER
+    else:
+        return dai.UsbSpeed.SUPER_PLUS
+
+
+def usbSpeedCorrection(s):
+    if s == 0:
+        return "UNKNOWN"
+    elif s == 1:
+        return "LOW"
+    elif s == 2:
+        return "FULL"
+    elif s == 3:
+        return "HIGH"
+    elif s == 4:
+        return "SUPER"
+    else:
+        return "SUPER_PLUS"
+
+
+def macCorrectFormat(s):
+    ret = ""
+    for x in s:
+        ret += str(hex(x)).strip("0x").upper() + ":"
+    return ret[:-1]
+
+
+def unlockConfig(window, devType):
+    if devType == "Poe":
+        window['staticIp'].update(disabled=False)
+        window['staticMask'].update(disabled=False)
+        window['staticGateway'].update(disabled=False)
+        window['dns'].update(disabled=False)
+        window['dnsAlt'].update(disabled=False)
+        window['networkTimeout'].update(disabled=False)
+        window['mac'].update(disabled=False)
+        window['staticBut'].update(disabled=False)
+        window['dynamicBut'].update(disabled=False)
+    else:
+        window['usbTimeout'].update(disabled=False)
+        window['usbSpeed'].update(disabled=False)
+    window['Flash newest version'].update(disabled=False)
+    window['Flash configuration'].update(disabled=False)
+    window['Factory reset'].update(disabled=False)
+    # window['Reset configuration'].update(disabled=False)
+
+
+def lockConfig(window):
+    window['staticIp'].update(disabled=True)
+    window['staticMask'].update(disabled=True)
+    window['staticGateway'].update(disabled=True)
+    window['dns'].update(disabled=True)
+    window['dnsAlt'].update(disabled=True)
+    window['networkTimeout'].update(disabled=True)
+    window['mac'].update(disabled=True)
+    window['staticBut'].update(disabled=True)
+    window['dynamicBut'].update(disabled=True)
+    window['usbTimeout'].update(disabled=True)
+    window['usbSpeed'].update(disabled=True)
+    window['Flash newest version'].update(disabled=True)
+    window['Flash configuration'].update(disabled=True)
+    window['Factory reset'].update(disabled=True)
+    window['Reset configuration'].update(disabled=True)
+
+    window.Element('staticIp').update("")
+    window.Element('staticMask').update("")
+    window.Element('staticGateway').update("")
+    window.Element('dns').update("")
+    window.Element('dnsAlt').update("")
+    window.Element('networkTimeout').update("")
+    window.Element('mac').update("")
+    window.Element('usbTimeout').update("")
+    window.Element('usbSpeed').update("")
+    window.Element('devName').update("-name-")
+    window.Element('newBoot').update("-version-")
+    window.Element('currBoot').update("-version-")
+
+
+def getDevices(window, devices):
+    listedDevices = []
+    devices.clear()
+    deviceInfos = dai.Device.getAllAvailableDevices()
```

Comment:
Use `dai.XLinkConnection.getAllConnectedDevices()` and print the state along them, just so this also serves as listing all devices (no matter the state)

Reply:
DONE:
- renamed file and moved to ```ustilities/device_manager.py``` (everything is now in 1 file)
- GUI now has 2 pages (```About``` and ```Configuration settings```)
- GUI also informs dai.__version__ and dai.__commmit__
- Added button ```Flash DAP``` that runs ```flashDepthaiApplicationPackage``` function
- Changed device finding from ```dai.Device.getAllAvailableDevices()``` to ```dai.XLinkConnection.getAllConnectedDevices()```
That is everything?